doc/arch/015-online-patient-lookup.md

```diff
@@ -0,0 +1,121 @@
+# Online Patient Lookup
+
+## Status
+
+Proposed on 2020-04-12
+
+## Context
+
+Want want to ensure that a patient’s medical records within the state
+are always available to the nurse for treatment if they are connected
+to the internet. See related
+[PRD](https://docs.google.com/document/d/1q6cppByQULfh3_mMXC4BJpiNN9Uc_awA6rreeEtUBaM/edit#)
+for more details on the feature and related specifications.
+
+The lookup will be a new API on the server that the mobile app will
+call. The following aspects of the problem are addressed in this ADR:
+
+- API contract
+- Data retention
+- Access restrictions
+- Audit logging
+- Rate limiting
+
+## Decision
+### Data retention
+There are two broad kinds of data retention on the app at a facility:
+1. Permanent: patient is within sync criteria: the same block, has an
+   appointment, or is assigned to the facility
+2. Temporary: patient is not within the sync criteria
+
+In order to have a better control of the retention, we will send this
+information from the server. This will be a static number to begin
+with, but can later vary depending on the state, country or other
+factors.
+
+We will implement temporary retention in the app with this
+feature. With every patient retrieved via the lookup API, we will:
+
+- ensure the data is synced to the server
+- delete the data after the retention time period has passed
+
+< @mobile-devs: please add details on possible approaches here >
+
+Note that the retention time period is not strict. i.e,
+it is okay for the record to stay on the phone for 2h if the retention
+period is 1h. We can perform deletions at a convenient time as long as
+the deletion happens deterministically.
+
+_Alternatively_: we can choose to hard-code the retention period on
+the app.
+
+### API contract
+For the request, we will use the endpoint `GET`:
+https://api.simple.org/api/v4/patient/identifier/, where `identifier`
+is any valid patient business identifier.
```

Comment:
How will the client specify the identifier type? eg. BP passport vs National Health ID. If we have thoughts about it, shall we put it in the ADR, since it's covering the API contract?

Reply:
I intended for the identifier to be of any type (and hence unspecified), since even the android app might not be in a good place to discern it. It's also possible for us to have the same UUID across Simple BP Passports and another ID.

When we return the patients, they'll have the identifier and its type in the payload in case the app needs it.

Good call though, I'll add this to the ADR.

Reply:
If I understand this correctly, it sounds like there are at least two types of valid business identifiers: BP passport and National Health ID. And it sounds like the client doesn't know which one it is, so we search for both and return matches for both. So duplicate matches would be very rare, correct? I think it would be good to get these details into the ADR because it would not have been clear what the "business identifiers" are or why there can be multiple matches.

Reply:
There can also be multiple matches because of duplicate registrations, which is actually most of what we've actually seen. Collisions across identifier types are just hypothetical atm. I'll call out both of these in the ADR.

Reply:
Added details around the type of identifier not being specified in the API.